src/dmd/tokens.d

```diff
@@ -715,24 +718,22 @@ extern (C++) struct Token
         }
     }
 
-    extern (D) private __gshared Token* freelist = null;
+    @disable this();
```

Comment:
I think a better solution is to move the constructor, alloc(), and free into the lexer. That way module tokens will be a leaf rather than part of another circle.

Reply:
Yes, `alloc` is already moved. I’ll try and move `free` as well. I tried to avoid changing the API, but it turns out I had to change it in other places instead.

Reply:
Fixed.